lib/auth/auth_with_roles.go

```diff
@@ -1229,6 +1229,13 @@ func (a *ServerWithRoles) GetPluginData(ctx context.Context, filter types.Plugin
 			}
 		}
 		return a.authServer.GetPluginData(ctx, filter)
+
+	case types.KindUser:
+		// TODO: Put correct auth checks here
```

Comment:
> 1. lib/auth/auth_with_roles.go: I am not sure how to correctly perform auth checks in UpdatePluginData() and GetPluginData(). We must ensure not only that a user has access to "UserPluginData", but that a user has access to a specific user in scope (himself).

It should be fine as is. `plugin_data` is a special permission which gives permission to update plugin data on any resource, without requiring update permissions for the resource in question. See https://github.com/gravitational/teleport/pull/6210.
```suggestion
```